awswrangler/postgresql.py

```diff
@@ -73,6 +74,41 @@ def _create_table(
     cursor.execute(sql)
 
 
+def _iterate_server_side_cursor(
+    sql: str,
+    con: Any,
+    chunksize: int,
+    index_col: Optional[Union[str, List[str]]],
+    params: Optional[Union[List[Any], Tuple[Any, ...], Dict[Any, Any]]],
+    safe: bool,
+    dtype: Optional[Dict[str, pa.DataType]],
+    timestamp_as_object: bool,
+) -> Iterator[pd.DataFrame]:
+    with con.cursor() as cursor:
+        sscursor_name: str = f"c_{uuid.uuid4().hex}"
+        cursor_args = _db_utils._convert_params(f"DECLARE {sscursor_name} CURSOR FOR {sql}", params)
+        cursor.execute(*cursor_args)
+
+        try:
+            while True:
+                cursor.execute(f"FETCH FORWARD {chunksize} FROM {sscursor_name}")
+                records = cursor.fetchall()
```

Comment:
Pg8000 is not fully DB API 2.0 - compliant with `fetchmany()` fetching all results but returning only x. This PR is a workaround to allow fetching only specific amount of results using server-side cursors.

Ultimately we'd like pg8000 to add full support for `fetchmany()` or add `SSCursor` implementation similar to MySQL, hiding the implementation details, but until it's there this workaround addresses the problem. I might do a PR to pg8000 later to address this.

Reply:
Good, idea, thanks! Added